src/misc.ts

```diff
@@ -0,0 +1,92 @@
+//
+// Types
+//
+
+/**
+ * Makes every specified property of the specified object type mutable.
+ *
+ * @template T - The object whose readonly properties to make mutable.
+ * @template TargetKey - The property key(s) to make mutable.
+ */
+export type Mutable<
+  T extends Record<string, unknown>,
+  TargetKey extends string
+> = {
+  -readonly [Key in keyof Pick<T, TargetKey>]: T[Key];
+} &
+  {
+    [Key in keyof Omit<T, TargetKey>]: T[Key];
+  };
+
+/**
+ * Useful for representing some value that _migh_ be present and / or complete.
+ *
+ * @template T - The value that might be present or complete.
+ */
+export type Maybe<T> = Partial<T> | null | undefined;
+
+/**
+ * Like {@link Array}, but always non-empty.
+ *
+ * @template T - The non-empty array member type.
+ */
+export type NonEmptyArray<T> = [T, ...T[]];
+
+/**
+ * A JavaScript object that is not `null`, a function, or an array. The object
+ * can still be an instance of a class.
+ */
+export type RuntimeObject = Record<number | string | symbol, unknown>;
+
+//
+// Type Guards
+//
+
+/**
+ * A {@link NonEmptyArray} type guard.
+ *
+ * @template T - The non-empty array member type.
+ * @param value - The value to check.
+ * @returns Whether the value is a non-empty array.
+ */
+export function isNonEmptyArray<T>(value: T[]): value is NonEmptyArray<T> {
+  return Array.isArray(value) && value.length > 0;
+}
+
+/**
+ * Type guard for "nullishness".
+ *
+ * @param value - Any value.
+ * @returns `true` if the value is null or undefined, `false` otherwise.
+ */
+export function isNullOrUndefined(value: unknown): value is null | undefined {
+  return value === null || value === undefined;
```

Comment:
You could simplify this as:
```suggestion
  return value == null;
```
But for clarity it might be better to explicitly check both.

Reply:
Yeah, I have avoided learning how `==` works thus far and I'm not about to do it now! :smile: